finalize: Try delete leftover MSET9 ID1

Naim2000 opened the pull request.

Naim2000 committed:
finalize: Try delete leftover MSET9 ID1

Naim2000 committed:
Update finalize.gm9

Naim2000 commented:
Should work now but like i can't really test it yk

lilyuwuu commented on finalize/finalize.gm9:
```diff
@@ -23,8 +23,19 @@ if find "0:/Nintendo 3DS/$[SYSID0]/????????????????????????????????_user-id1" NU
+
+# What is the NULL parameter here for? If I changed it, would it become a variable holding the name for the found folder?
+# Where can I read up on gm9script?
+# It was here https://github.com/eiiiiix/EixMode9/blob/master/HelloScript.gm9
```
should use [upstream's helloscript](https://github.com/d0k3/GodMode9/blob/master/resources/sample/HelloScript.gm9) rather than random fork (content should be identical)

Naim2000 replied:
have to admit i searched up 'godmode9 script engine' on google and that was like the first thing

lilyuwuu commented on finalize/finalize.gm9:
```diff
@@ -23,8 +23,19 @@ if find "0:/Nintendo 3DS/$[SYSID0]/????????????????????????????????_user-id1" NU
+else if find "0:/Nintendo 3DS/$[SYSID0]/????????????????????????sdmc??b9 MSET9ID1 # Else if because I don't want the user to power on and have the system create another ID1
+	# Silently delete it.
+	rm -s $[MSET9ID1]
+	# -o = optional, i.e. don't sh*t itself when it fails. I don't want this to fail.
```
if it does fail, it should ideally be surrounded by an if clause to give the user more info rather than gm9's "failed lmao"

lilyuwuu replied:
oh right by the way the quotation mark isn't closed on the if find line

Naim2000 replied:
About this.... how do I explain to the user that the hax ID1 didn't get deleted exactly

lilyuwuu replied:
most likely something along the lines of "Failed to delete hax ID1. This shouldn't happen. Ask for help [...]"

I am wondering whether GM9 is able to handle this through scripting though -- GM9 definitely can delete hax id1 in general but it seems like what's currently written here doesn't work (even though it should?)

On another note since this probably falls under gm9's protection scheme you most likely would need to prompt the user to allow key combos (so something like "hax id1 detected, please do the key combo on the next prompt")

lilyuwuu reviewed (commented):
to clarify, in what situation would someone end up in this scenario, and what folders would they end up having in ID0?

irowiki commented:
Does this also rename the real ID1 and remove _user-id1 from it?

irowiki commented:
> to clarify, in what situation would someone end up in this scenario, and what folders would they end up having in ID0?

If someone doesn't remove MSET9 they'll have the hax ID1 and the normal ID1 with _user-id1 appended to it

Naim2000 commented:
> to clarify, in what situation would someone end up in this scenario, and what folders would they end up having in ID0?

Happens to chrome OS users a lot. The original ID1 gets renamed but the hax id1 never gets deleted

lilyuwuu commented:
that would be an... interesting scenario. I'm mainly concerned with whether GodMode9 actually mounts A:/ in this scenario (in theory it *should* since proper id0-id1 combination exists and the hax id1 is just ignored by gm9) but not totally sure. If it doesn't get mounted then we probably need to reboot

Naim2000 commented:
> to clarify, in what situation would someone end up in this scenario, and what folders would they end up having in ID0?

Also, it'll be their normal ID1 (without the _user-id1 suffix!) and the hax id1. Example https://discord.com/channels/196618637950451712/196635695958196224/1211461993119096863

Naim2000 committed:
fix unterminated string

lilyuwuu commented:
does not appear to work (tested on console)
in gm9 `else if` is `elif`

Naim2000 committed:
it's elif

irowiki commented:
Would need some sort of sanity checking to see if _user-id1 is still present on the normal ID1 for those that just didn't remove it, otherwise if you remove the hax ID1 and leave the other one with _user-id1, 3DS will just make a new ID1 yeah? If GM9 can't rename it at that point it would need some sort of error flag for manual intervention.

lilyuwuu commented:
If \_user-id1 is present at all the script will halt. The only way it can get to the deleting hax id1 part is if \_user-id1 is not present

Naim2000 committed:
User friendly handling

J0n-b0 committed:
Fixed it

Fixed issues mentioned in pull request. Requires blue gm9 prompt and a SD eject which may scare some users.

Naim2000 committed:
Merge pull request #1 from J0n-b0/patch-1

Fixed MSET9 Stuff

J0n-b0 commented:
Works fine now, with support for removal of both _user-id1 and MSET9 ID1.

lilyuwuu commented:
bdf42b5
